src/modules/clusters/cluster-list/cluster-list.js

```diff
@@ -433,34 +526,54 @@
             return matches;
         }
 
-        function _applyFilters(filters) {
-            vm.filteredClusterList = [];
+        function _applyFilters(filters, head, type) {
+
+            head.filteredClusterList = [];
+            if (filters && filters.length > 0) {
+                head.clusterList.forEach(function(item) {
+                    if (_matchesFilters(item, filters)) {
+                        head.filteredClusterList.push(item);
+                    }
+                });
+            } else {
+                head.filteredClusterList = head.clusterList;
+            }
+            head.filterConfig.resultsCount = head.filteredClusterList.length;
+        }
+
+        function _applyHostFilters(filters, head, type) {
+            head.filteredHostList = [];
             if (filters && filters.length > 0) {
-                vm.clusterList.forEach(function(item) {
+                head.hosts.forEach(function(item) {
                     if (_matchesFilters(item, filters)) {
-                        vm.filteredClusterList.push(item);
+                        head.filteredHostList.push(item);
                     }
                 });
             } else {
-                vm.filteredClusterList = vm.clusterList;
+                head.filteredHostList = head.hosts;
             }
-            vm.filterConfig.resultsCount = vm.filteredClusterList.length;
+            head.filterConfig.resultsCount = head.filteredHostList.length;
         }
 
-        function _filterChange(filters) {
-            vm.filtersText = "";
-            vm.filters = filters;
+        function _filterConf(filters, head, type) {
+            head.filtersText = "";
```

Comment:
@a2batic Instead of head, it should be named as "context"

Reply:
Ack